### PR TITLE
Cleanup docstrings for `gammapy.irf`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue https://github.com/gammapy/gammapy/issues/4811, specifically for `gammapy.irf`

### CODE REVIEW

File: gammapy/irf/core.py [hunk 1]
```diff
@@ -122,12 +122,12 @@ def data(self):
...
     @data.setter
     def data(self, value):
-        """Set data
+        """Set data.
 
         Parameters
         ----------
-        value : array-like
-            Data array
+        value : `~np.ndarray`
```
Comment (registerrier) [inline gammapy/irf/core.py:129]:
```suggestion
        value : `~numpy.ndarray`
```

File: gammapy/irf/core.py [hunk 2]
```diff
@@ -512,18 +512,25 @@ def to_table_hdu(self, format="gadf-dl3"):
...
         return fits.BinTableHDU(self.to_table(format=format), name=name)
 
     def to_hdulist(self, format="gadf-dl3"):
-        """"""
+        """
+        Write the HDU list.
+
+        Parameters
+        ----------
+        format : {"gadf-dl3"}, optional
```
Comment (registerrier) [inline gammapy/irf/core.py:531]: Maybe specify `optional` in other methods using the `format` argument.

File: gammapy/irf/edisp/kernel.py [hunk 3]
```diff
@@ -77,9 +77,9 @@ def pdf_in_safe_range(self, lo_threshold, hi_threshold):
         Parameters
         ----------
         lo_threshold : `~astropy.units.Quantity`
-            Low reco energy threshold
+            Low reconstructed energy threshold.
```
Comment (registerrier) [inline gammapy/irf/edisp/kernel.py:80]: If you change here, you will have to change everywhere. Is it worth it?

File: gammapy/irf/edisp/map.py [hunk 4]
```diff
@@ -26,15 +26,16 @@ class EDispMap(IRFMap):
...
-        the input Energy Dispersion Map. Should be a Map with 2 non spatial axes.
+        The input Energy Dispersion Map. Should be a Map with 2 non spatial axes.
         migra and true energy axes should be given in this specific order.
     exposure_map : `~gammapy.maps.Map`, optional
         Associated exposure map. Needs to have a consistent map geometry.
 
     Examples
     --------
-    .. doctest::
+    ::
```
Comment (registerrier) [inline gammapy/irf/edisp/map.py:36]: is the code executed in doctest this way?

**Review by registerrier — COMMENTED** [review]
Thanks @Astro-Kirsty . See inline comments.

File: gammapy/irf/edisp/kernel.py (hunk 3 above)
Comment (Astro-Kirsty) [reply gammapy/irf/edisp/kernel.py:80]: Most of the time it is actually 'reconstructed' in the docstring. So there is currently inconsistency. 
I would say change to 'reconstructed' where necessary.

File: gammapy/irf/edisp/map.py (hunk 4 above)
Comment (Astro-Kirsty) [reply gammapy/irf/edisp/map.py:36]: I'm not sure why a doctest was included here, they are not included in other Examples sections?

File: gammapy/irf/psf/parametric.py [hunk 5]
```diff
@@ -299,11 +301,11 @@ class EnergyDependentMultiGaussPSF(ParametricPSF):
...
     ----------
     axes : list of `MapAxis`
-        Required axes are ["energy_true", "offset"]
+        Required axes are ["energy_true", "offset"].
     data : `~numpy.recarray`
-        Data array
+        Data array.
     meta : dict
-        Meta data
+        Meta data.
```
Comment (cgalelli) [inline gammapy/irf/psf/parametric.py:308]:
```suggestion
        Metadata.
```

**Review by cgalelli — COMMENTED** [review]
Looks good @Astro-Kirsty ! The usual handful of inline comments and suggestions

**Review by cgalelli — APPROVED** [review]
Thanks, this looks good!

**Pull request merged**